src/lib/InstantSearch.js

```diff
@@ -3,13 +3,11 @@
 import algoliasearch from 'algoliasearch/src/browser/builds/algoliasearchLite.js';
 import algoliasearchHelper from 'algoliasearch-helper';
 import forEach from 'lodash/forEach';
-import mergeWith from 'lodash/mergeWith';
-import union from 'lodash/union';
-import isPlainObject from 'lodash/isPlainObject';
 import { EventEmitter } from 'events';
 import urlSyncWidget from './url-sync.js';
 import version from './version.js';
 import createHelpers from './createHelpers.js';
+import { enhanceConfiguration } from './utils.js';
```

Comment:
I'd move that either in its own files or in a dedicated package that is less "everything go". What about "lifecycle.js"?

Reply:
Not the scope of the PR, we should keep this on mind and have it in another one or with the instantsearch-core refactoring.

Reply:
I don't agree on this one. You are moving some code out of a file, but utils.js is not the good file for that.

Reply:
Which other functions could we put inside lifecycle.js do you think?

Reply:
I definitely didn't give much information here...

Moving this function into utils is not a good move. It makes the situation worse, because the package didn't have a clear purpose and this function is not even consistent with the one that are already there.

That being putting in a separate file doesn't really make for a clear better situation either.

What about keeping this function in the package where it was and exporting it?

Reply:
Yes, good point @bobylito having the function exported from the original file did not crossed my mind.